tests/test_splicing.py

```diff
@@ -276,3 +276,44 @@ def test_commit_crash_splice(node_factory, bitcoind):
     # Check that the splice doesn't generate a unilateral close transaction
     time.sleep(5)
     assert l1.db_query("SELECT count(*) as c FROM channeltxs;")[0]['c'] == 0
+
+
+@pytest.mark.openchannel('v1')
+@pytest.mark.openchannel('v2')
+@unittest.skipIf(TEST_NETWORK != 'regtest', 'elementsd doesnt yet support PSBT features we need')
+def test_splice_stuck_htlc(node_factory, bitcoind, executor):
+    l1, l2, l3 = node_factory.line_graph(3, wait_for_announce=True, opts={'experimental-splicing': None})
+
+    l3.rpc.dev_ignore_htlcs(id=l2.info['id'], ignore=True)
+    
+    inv = l3.rpc.invoice(10000000, '1', 'no_1')
+    executor.submit(l1.rpc.pay, inv['bolt11'])
+    l3.daemon.wait_for_log('their htlc 0 dev_ignore_htlcs')
+
+    # Now we should have a stuck invoice between l1 -> l2
+
+    chan_id = l1.get_channel_id(l2)
+
+    # add extra sats to pay fee
+    funds_result = l1.rpc.fundpsbt("109000sat", "slow", 166, excess_as_change=True)
+
+    result = l1.rpc.splice_init(chan_id, 100000, funds_result['psbt'])
+    result = l1.rpc.splice_update(chan_id, result['psbt'])
+    result = l1.rpc.signpsbt(result['psbt'])
+    result = l1.rpc.splice_signed(chan_id, result['signed_psbt'])
+
+    l2.daemon.wait_for_log(r'CHANNELD_NORMAL to CHANNELD_AWAITING_SPLICE')
+    l1.daemon.wait_for_log(r'CHANNELD_NORMAL to CHANNELD_AWAITING_SPLICE')
+
+    mempool = bitcoind.rpc.getrawmempool(True)
+    assert len(list(mempool.keys())) == 1
+    assert result['txid'] in list(mempool.keys())
+
+    bitcoind.generate_block(6, wait_for_mempool=1)
+
+    l2.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
+    l1.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
+
+    # Check that the splice doesn't generate a unilateral close transaction
+    time.sleep(5)
```

Comment:
is it better wait for "Channel ready for use. Channel announced." log or just an update from the channel? 

I am not sure that this sleep time is enough in all the machine

Reply:
Ya I agree using sleep is annoying in the testing framework.

My reason for using it in this case is that I wanted to be extra certain this doesn't happen, as a delayed event might cause a unilateral close later -- after our log message.

> I am not sure that this sleep time is enough in all the machine

The event should happen quite quickly, 1<s. Five seconds was chosen as a number to be assuredly well past that.

Reply:
>The event should happen quite quickly, 1<s. Five seconds was chosen as a number to be assuredly well past that.

Not sure that in all machines with VALGRID enabled this is true. It is better to wait for another kind of log, like a channel update or somethings

Reply:
Yeah, we do this some time: it's hard when you're waiting for something which *doesnt* happen.  It's fine.